### ISSUE TITLE
The whole app restated when using hot reload

### ISSUE THREAD

**Issue by TareqAmenahDev** [aissat/easy_localization#468]
I am using easy_localization for translations in my Flutter app. everything is fine but while developing every time I use hot reload the whole app is restarted.

this is my question on stackoverflow:
https://stackoverflow.com/questions/71716531/easy-localization-restart-the-whole-app-every-time-i-using-hot-reload

**Comment by konradmietelski** [comment]
@TareqAmenahDev What was source of the problem? How did you resolve it?

**Comment by konradmietelski** [comment]
For anyone experiencing this issue, it is almost certainly not related to `easy_localization`.

TL;DR: In my `initialRoute` named `SplashView`, I had some conditional logic to determine which page should be rendered. Every time hot reload was called, my logic was executed, and the user was redirected to `HomeView`.

I updated my codebase to the stable version of Dart/Flutter (3.32 as of today) and updated the dependencies, but the problem persisted. I had a side project without `easy_localization`, so I installed it there, and there was no problem with hot reload. After some testing and further investigation, I discovered that the issue must be related to routing, not to state preservation or rendering.

I recreated the basic structure of the project from scratch to debug it properly. In the new project, I added and configured every pub package I had in the original project. Step by step, I added a package, configured it, ran `flutter clean`, and then `flutter run` - I repeated this process multiple times, ending up with all of packages in both projects. However, hot reload was working just fine in the "debug" project but not in the other one.

I compared the configurations and base files of both projects (the `/android` folder, `main.dart`, etc.). I noticed a couple of differences, so I tried to rule them out one by one. I checked `navigatorKey` and `restorationScopeId`, as suggested by some online sources, but with no success.

But then I did notice, that `initialRoute` (used in both projects) is handled differently. In my „debug” project this Route was simple - it was just `HomeView`. But in my „problematic” project there was some more logic inside (which - as for now - I believe is very bad practice ;).
My `initialRoute` in „problematic” project is named `SplashView` and it triggers some data syncing and extra processing, after which it is redirecting to proper page. By proper page I mean for most cases - start page. So hot reload IS calling/using initialRoute and if route defined there is using Navigator to point user to another view - hot reload will follow.


Then I noticed that `initialRoute` (used in both projects) is handled differently. In my "debug" project, this route was simple - it was just `HomeView`. In my "problematic" project, however, there was additional logic involved (which, for now, I believe is very bad practice). My `initialRoute` in the "problematic" project is named SplashView, and it triggers some data syncing and extra processing, after which it redirects to the appropriate page. By "appropriate page" I mean the `HoveView` (in most cases). So, hot reload is calling/using `initialRoute`, and if the route defined there uses the `Navigator` to point the user to another view, hot reload will follow.
